src/handlers/uniswapLiquidity.ts

```diff
@@ -1,301 +1,140 @@
+
+  const firstSwapDetails = await getQuote(
+    network,
+    fromTokenAddress,
+    buyToken,
+    fromAmount
   );
-  const v2TokensCall = getLiquidityInfoV2(
-    chainId,
-    accountAddress,
-    v2Tokens,
-    pairs
-  );
-  const [v1Results, v2Results] = await Promise.all([
-    v1TokensCall,
-    v2TokensCall,
-  ]);
-  return { ...v1Results, ...v2Results };
-};
 
-const getLiquidityInfoV2 = async (
-  chainId: ChainId,
-  accountAddress: string,
-  liquidityTokens: ParsedAddressAsset[],
-  pairs: Record<string, Asset>
-): Promise<Record<string, LiquidityInfo>> => {
-  const promises = map(liquidityTokens, async lpToken => {
-    try {
-      const liquidityPoolAddress = lpToken.address;
-      const pair = new Contract(
-        liquidityPoolAddress,
-        IUniswapV2PairABI,
-        web3Provider
-      );
-
-      const token0AddressCall = pair.token0();
-      const token1AddressCall = pair.token1();
-      const pairReservesCall = pair.getReserves();
-      const totalSupplyCall = pair.totalSupply();
-
-      const [
-        token0AddressResult,
-        token1AddressResult,
-        pairReservesResult,
-        totalSupplyResult,
-      ] = await Promise.all([
-        token0AddressCall,
-        token1AddressCall,
-        pairReservesCall,
-        totalSupplyCall,
-      ]);
-
-      const token0Address = token0AddressResult.toString();
-      const token1Address = token1AddressResult.toString();
-
-      const lpTokenBalance = convertAmountToRawAmount(
-        lpToken?.balance?.amount || 0,
-        lpToken?.decimals
-      );
-
-      const [token0ReserveBn, token1ReserveBn] = pairReservesResult;
-      const token0Reserve = token0ReserveBn.toString();
-      const token1Reserve = token1ReserveBn.toString();
-      const totalSupply = totalSupplyResult.toString();
-
-      const token0 = await getTokenDetails(chainId, token0Address, pairs);
-      const token1 = await getTokenDetails(chainId, token1Address, pairs);
-
-      const token0Balance = convertRawAmountToDecimalFormat(
-        divide(multiply(token0Reserve, lpTokenBalance), totalSupply),
-        token0.decimals
-      );
-
-      const token1Balance = convertRawAmountToDecimalFormat(
-        divide(multiply(token1Reserve, lpTokenBalance), totalSupply),
-        token1.decimals
-      );
-
-      return {
-        ...lpToken,
-        tokens: [
-          {
-            balance: token0Balance,
-            ...token0,
-          },
-          {
-            balance: token1Balance,
-            ...token1,
-          },
-        ],
-        totalSupply: convertRawAmountToDecimalFormat(totalSupply),
-      };
-    } catch (error) {
-      logger.log('error getting uniswap info', error);
-      return null;
-    }
-  });
-
-  const results = await Promise.all(promises);
-  const updatedResults = compact(results);
-  return keyBy(updatedResults, result => result.address);
+  if (!firstSwapDetails) return null;
+
+  const minPoolTokens = '1';
```

Comment:
Note: This was copying the behavior of Zapper but we should calculate actual min pool token